tests/pubsub/check_pubsub_subscribe_rt_levels.c

```diff
@@ -345,114 +343,122 @@ START_TEST(SetupInvalidPubSubConfig) {
-    ck_assert_int_eq(retVal, UA_STATUSCODE_GOOD);
-
-    readerConfig.subscribedDataSet.subscribedDataSetTarget.targetVariablesSize = 1;
-    readerConfig.subscribedDataSet.subscribedDataSetTarget.targetVariables     = (UA_FieldTargetVariable *)
-        UA_calloc(readerConfig.subscribedDataSet.subscribedDataSetTarget.targetVariablesSize, sizeof(UA_FieldTargetVariable));
-
-    /* For creating Targetvariable */
-    UA_FieldTargetDataType_init(&readerConfig.subscribedDataSet.subscribedDataSetTarget.targetVariables[0].targetVariable);
-    readerConfig.subscribedDataSet.subscribedDataSetTarget.targetVariables[0].targetVariable.attributeId  = UA_ATTRIBUTEID_VALUE;
-    readerConfig.subscribedDataSet.subscribedDataSetTarget.targetVariables[0].targetVariable.targetNodeId = subNodeId;
-
-    retVal = UA_Server_addDataSetReader (server, readerGroupIdentifier, &readerConfig,
-                                          &readerIdentifier);
-    ck_assert_int_eq(retVal, UA_STATUSCODE_GOOD);
-    UA_NodeId readerIdentifier2;
-    retVal = UA_Server_addDataSetReader (server, readerGroupIdentifier, &readerConfig,
-                                          &readerIdentifier2);
-    ck_assert_int_eq(retVal, UA_STATUSCODE_GOOD);
-    UA_UadpDataSetReaderMessageDataType_delete(dataSetReaderMessage);
-
-    UA_FieldTargetDataType_clear(&readerConfig.subscribedDataSet.subscribedDataSetTarget.targetVariables[0].targetVariable);
-    UA_free(readerConfig.subscribedDataSet.subscribedDataSetTarget.targetVariables);
-    UA_free(readerConfig.dataSetMetaData.fields);
-    UA_Variant_clear(&variant);
-
-    ck_assert(UA_Server_freezeReaderGroupConfiguration(server, readerGroupIdentifier) == UA_STATUSCODE_BADNOTIMPLEMENTED); // Multiple DSR not supported
-
-    ck_assert(UA_Server_unfreezeReaderGroupConfiguration(server, readerGroupIdentifier) == UA_STATUSCODE_GOOD);
-    retVal = UA_Server_removeDataSetReader(server, readerIdentifier2);
-    ck_assert_int_eq(retVal, UA_STATUSCODE_GOOD);
-
-    ck_assert(UA_Server_freezeReaderGroupConfiguration(server, readerGroupIdentifier) == UA_STATUSCODE_BADNOTSUPPORTED); // DateTime not supported
-    ck_assert(UA_Server_freezeWriterGroupConfiguration(server, writerGroupIdent) == UA_STATUSCODE_BADNOTSUPPORTED); // Static value source only supported
-    ck_assert(UA_Server_unfreezeReaderGroupConfiguration(server, readerGroupIdentifier) == UA_STATUSCODE_GOOD);
-    ck_assert(UA_Server_unfreezeWriterGroupConfiguration(server, writerGroupIdent) == UA_STATUSCODE_GOOD);
+    UA_DataSetWriterConfig dataSetWriterConfig;
+    memset(&dataSetWriterConfig, 0, sizeof(UA_DataSetWriterConfig));
+    dataSetWriterConfig.name = UA_STRING("Test DataSetWriter");
+    dataSetWriterConfig.dataSetWriterId = 62541;
+    ck_assert(UA_Server_addDataSetWriter(server, writerGroupIdent, publishedDataSetIdent, &dataSetWriterConfig, &dataSetWriterIdent) == UA_STATUSCODE_BADCONFIGURATIONERROR);
```

Comment:
If the correct order of calls is used (`addDataSetWriter` after `addDataSetField`), the invalid pubsub configuration is already intercepted at this point, which makes the rest of the nest unnecessary.

Reply:
Added another test where the commented out codepaths for the reader are tested again.